core/templates/dev/head/App.ts

```diff
@@ -190,12 +191,13 @@ oppia.config([
     // Add an interceptor to convert requests to strings and to log and show
     // warnings for error responses.
     $httpProvider.interceptors.push([
-      '$q', '$log', 'AlertsService', function($q, $log, AlertsService) {
+      '$q', '$log', 'AlertsService', 'CsrfService',
+      function($q, $log, AlertsService, CsrfService) {
         return {
           request: function(config) {
             if (config.data) {
               config.data = $.param({
-                csrf_token: GLOBALS.csrf_token,
+                csrf_token: CsrfService.getToken(),
```

Comment:
What if this is called before we the request for CSRF finishes? We should return a promise from `.getToken()` that will resolve only if the token is already available.

Reply:
@Jamesjay4199 When you mark a comment as resolved, please always include an explanation (for the benefit of future developers/reviewers). Could you respond to @vojtechjelinek's comment?

Reply:
I think this will never be the case because the csrf token is only used for post requests and a post request is only sent based on the user interaction with the page, while the request gets sent much earlier

Reply:
It could theoretically happen though (e.g. if the server is overloaded for some reason). What will the user experience in this case, and what should they experience ideally?

Reply:
I've refactored it to use a promise.
@vojtechjelinek PTAL

Reply:
Why do you need to use jQuery? AngularJS has $q, try to use that for promises if possible.